### ISSUE TITLE
Allow custom rule/rule engine listeners in AbstractRuleEngine

### ISSUE THREAD

**Issue by elitacco** [j-easy/easy-rules#215]
After #157 there is no way to remove the default rule and rules engine listeners. It should be possible to create a rule engine with a custom list of rule listeners and rule engine listeners.

**Comment by okohub** [comment]
+1 for this one.

This was a breaking change for us. We are removing default listeners when decorating DefaultRulesEngine.

**Comment by zhhaojie** [comment]
> After #157 there is no way to remove the default rule and rules engine listeners. It should be possible to create a rule engine with a custom list of rule listeners and rule engine listeners.

yes, you can register your custom  list of rule listeners & rule engine listeners. but there is no way to remove the default listeners.

i wonder what's the reason you remove the defaults ?

**Comment by benas** [comment]
Default listeners will be removed in next major version (See #223).

I'm closing this issue as a duplicate of #223 .
